java/vector/src/main/java/org/apache/arrow/vector/LargeVarCharVector.java

```diff
@@ -120,12 +123,28 @@ public byte[] get(int index) {
    * @return Text object for non-null element, null otherwise
    */
   public Text getObject(int index) {
-    byte[] b = get(index);
-    if (b == null) {
+    assert index >= 0;
+    if (NULL_CHECKING_ENABLED && isSet(index) == 0) {
```

Comment:
Won't this use of NULL_CHECKING_ENABLED completely change the behavior of this method? Instead of returning null it will return Text(null)?

It's a matter for a different PR, but I don't see how NULL_CHECKING_ENABLED is useful in any possible way, because it's global to all recursive dependencies of anything using Arrow for any purpose.